File: BOOKWORM_PRO/Bookworm/src/main/java/com/example/Controller/AuthController.java
package com.example.Controller;

import com.example.Services.AuthService;
import com.example.dto.AuthResponse;
import com.example.dto.LoginRequest;
import com.example.dto.RegisterRequest;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/auth")
@CrossOrigin(origins = "http://localhost:5173")
public class AuthController {

    private final AuthService service;

    public AuthController(AuthService service) {
        this.service = service;
    }

    @PostMapping("/register")
    public String register(@RequestBody RegisterRequest request) {
        service.register(request);
        return "User Registered Successfully";
    }
    
    
    @PostMapping("/login")
    public AuthResponse login(@RequestBody LoginRequest request) {
        return service.login(request);
    }

//    @PostMapping("/login")
//    public AuthResponse login(@RequestBody LoginRequest request) {
//        return new AuthResponse(service.login(request));
//    }
}

File: BookWorm-NET/BookWormNET/Controllers/AuthController.cs
using BookWormNET.Data;
using BookWormNET.dto;
using BookWormNET.Services.Implementation;
using BookWormNET.Services.Interfaces;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.Logging;

namespace BookWormNET.Controllers
{
    [ApiController]
    [Route("api/auth")]
    public class AuthController : ControllerBase
    {
        private readonly IAuthService _authService;
        private readonly ILogger<AuthController> _logger;

        public AuthController(IAuthService authService, ILogger<AuthController> logger)
        {
            _authService = authService;
            _logger = logger;

            _logger.LogInformation("============AuthController hit============");
        }

        [HttpPost("login")]
        public async Task<IActionResult> Login(LoginRequestDto request)
        {

            _logger.LogInformation("Login action started for username/email: {Email}", request.Email);
            try
            {
                var result = await _authService.LoginAsync(request);

                _logger.LogInformation("Login successful for username/email: {Email}", request.Email);

                return Ok(result);
            }
            catch(Exception ex)
            {
                _logger.LogWarning(ex, "Login failed for username/email: {Email}", request.Email);
                return Unauthorized("Invalid credentials");
            }
        }

        [HttpPost("register")]
        public async Task<IActionResult> Register(RegisterDto request)
        {
            await _authService.RegisterAsync(request);
            return Ok("User registered successfully");
        }


        [HttpPost("register-admin")]
        public async Task<IActionResult> RegisterAdmin(AdminRegisterDto request)
        {
            await _authService.RegisterAdminAsync(request);
            return Ok("Admin registered successfully");
        }

    }

}
